pkg/bucketeer/cache/in_memory_cache.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 The Bucketeer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate mockgen -source=$GOFILE -package=mock -destination=./mock/$GOFILE
+package cache
+
+import (
+	"sync"
+	"time"
+)
+
+const (
+	defaultEvictionInterval = 0
+)
+
+type entry struct {
+	value      interface{}
+	expiration time.Time
+}
+
+type inMemoryCache struct {
+	entries          sync.Map
```

Comment:
I have a suggestion about the `sync.Map`: If we have a map with many updates, benchmark the performance of sync.Map with a mutex against a standard map with a mutex. We might find the standard map performs better in that case.

Reply:
Generally, `sync.Map` is a great choice for read-only or written once with frequent reads.

Reply:
The cache will be updated once per minute only when needed. It will be used more for reading than writing.
It should be fine for this case, but I can change it if you still have any concerns.

Reply:
I think it should be okay to update the in-memory cache every minute. 
We can keep this in mind if the `sync.Map` encounters performance issues in the future.

Reply:
The rest code LGTM!

Reply:
We have been using `sync.Map` in production to get the API from the cache with no performance issues.
But I'll keep monitoring it!
Thank you.